epsagon/custom_trace_data_test.go

```diff
@@ -0,0 +1,312 @@
+package epsagon_test
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const TestLabelKey = "random_key"
+
+func TestEpsagonCustomTraceFields(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Custom trace fields tests")
+}
+
+func waitForTrace(traceChannel chan *protocol.Trace, resourceName string) *protocol.Event {
+	var trace *protocol.Trace
+	receivedTrace := false
+	ticker := time.NewTicker(3 * time.Second)
+	for !receivedTrace {
+		select {
+		case trace = <-traceChannel:
+			func() {
+				Expect(len(trace.Events)).To(Equal(1))
+				if len(resourceName) > 0 {
+					Expect(trace.Events[0].Resource.Name).To(Equal(resourceName))
+				}
+				receivedTrace = true
+			}()
+		case <-ticker.C:
+			panic("timeout while receiving trace")
+		}
+	}
+	return trace.Events[0]
```

Comment:
If the ticker was timeout can we still access trace.Events[0]?

Reply:
if it was timeout then the func will panic so it won't be a problem :)